tests/test_debug/test_snapshot_manager.py

```diff
@@ -0,0 +1,151 @@
+        num = 123
+
+
+def func_multi_line(output_directory: str = None) -> None:
+    with SnapshotManager(output_directory=output_directory, include=("func_multi_line",)):
+        num = 123
+        some_string = "Hello, world"
+        num2 = 321
+        arr = [some_string, "string 123321"]
+
+
+def func_mutation(output_directory: str = None) -> None:
+    with SnapshotManager(output_directory=output_directory, include=("func_mutation",)):
+        num = 123
+        num = 321
+
+
+def func_for_loop(output_directory: str = None) -> None:
+    with SnapshotManager(output_directory=output_directory, include=("func_for_loop",)):
+        nums = [1, 2, 3]
+        for i in range(len(nums)):
+            nums[i] = nums[i] + 1
+
+
+def func_if_else(output_directory: str = None) -> None:
+    with SnapshotManager(output_directory=output_directory, include=("func_if_else",)):
+        num = 10
+        if num > 5:
+            result = "greater"
+        else:
+            result = "lesser"
+
+
+def func_while(output_directory: str = None) -> None:
+    with SnapshotManager(output_directory=output_directory, include=("func_while",)):
+        num = 0
+        while num < 3:
+            num += 1
+
+
+def func_with_args():
+    with SnapshotManager(
+        include=("func_with_args",), memory_viz_args=["--roughjs-config", "seed=12345"]
+    ):
+        flag = "not --output"
+
+
+def assert_output_files_match(output_path: str, snapshot: Snapshot, function_name: str):
+    actual_svgs = {}
+    files = os.listdir(output_path)
+    for file in files:
+        actual_path = os.path.join(output_path, file)
+        with open(actual_path) as actual_file:
+            actual_svg = actual_file.read()
+            actual_svgs[file] = actual_svg
+    snapshot.assert_match_dir(actual_svgs, function_name)
+
+
+def func_no_memory_viz_args():
+    with SnapshotManager(include=("func_no_memory_viz_args",)):
+        s = "Hello"
+
+
+@pytest.fixture(scope="function")
+def test_directory(func_name, snapshot):
+    snapshot.snapshot_dir = SNAPSHOT_DIR
+    actual_dir = os.path.join(TEST_RESULTS_DIR, func_name)
+    shutil.rmtree(actual_dir, ignore_errors=True)
+    os.makedirs(actual_dir, exist_ok=True)
+    yield actual_dir
+    shutil.rmtree(actual_dir, ignore_errors=True)
+
+
+@pytest.fixture(scope="function")
+def setup_curr_dir_testing(snapshot):
+    snapshot.snapshot_dir = SNAPSHOT_DIR
+    file_name = "snapshot-0.svg"
+    if os.path.exists(file_name):
+        os.remove(file_name)
+    yield
+    os.remove(file_name)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 10), reason="requires Python 3.10 or higher")
+def test_func_with_non_output_flags(snapshot, setup_curr_dir_testing):
+    func_with_args()
+
+    with open("snapshot-0.svg") as actual_file:
+        snapshot.assert_match_dir({"snapshot-0.svg": actual_file.read()}, func_with_args.__name__)
+
+
+functions_to_test = [
```

Comment:
Don't provide a separate definition, put this inside the `parameterize` call. (This file has a lot of indirect definitions already, so we don't need to add more.)

Also, you don't need to generate `func.__name__` in this list, just pass in the function itself.